src/Features/Core/Portable/Completion/CompletionService.cs

```diff
@@ -59,6 +59,31 @@ public static CompletionService GetService(Document document)
             return false;
         }
 
+        /// <summary>
+        /// Returns true if the character recently inserted or deleted in the text should trigger completion.
+        /// </summary>
+        /// <param name="project">The project containing the document and text</param>
+        /// <param name="text">The document text to trigger completion within </param>
+        /// <param name="caretPosition">The position of the caret after the triggering action.</param>
+        /// <param name="trigger">The potential triggering action.</param>
+        /// <param name="roles">Optional set of roles associated with the editor state.</param>
+        /// <param name="options">Optional options that override the default options.</param>
+        /// <remarks>
+        /// We pass the project here to retrieve information about the <see cref="Project.AnalyzerReferences"/>,
+        /// <see cref="WorkspaceKind"/> and <see cref="Project.Language"/> which are fast operations.
+        /// It should not be used for syntactic or semantic operations.
+        /// </remarks>
```

Comment:
Can we consider making this a public api instead of internal? It doesn't seem to be using any internal bits.

Reply:
I'm open to it in a followup - how are you thinking of using it?  the same case where we are to avoid roundtripping the sourcetext?

Reply:
Correct. Today omnisharp gets the document, gets the sourcetext for the document, and then calls http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/Completion/CompletionService.cs,5744063638701bd9. This seems like the same scenario.